assets/js/theme/gift-certificate.js

```diff
@@ -168,16 +167,15 @@ export default class GiftCertificate extends PageManager {
 
             const modal = defaultModal();
             const previewUrl = `${$(event.currentTarget).data('previewUrl')}&${$purchaseForm.serialize()}`;
+            const frameOptions = {
+                height: '350px',
```

Comment:
I don't like the idea of hardcoding the height and width of the iframe. Since this might break mobile devices. We should look into if we can calculate it runtime based on the modal and assign the same height and width.
Thoughts ?

Reply:
I agree that under normal circumstances, we would not want to set the width and height of an iframe with fixed pixel values. However, since the content being loaded in the frame will only be the gift certificate preview, which is a table with a fixed width of 560px, I felt this was an acceptable situation to use a fixed width. The gift certificate template itself is not responsive, so we don't need to worry about its width changing across various screens.

As it stands, the current implementation loads the gift certificate preview in with the ability to side-scroll on mobile since the preview is wider than the mobile viewport.

As for the fixed height, this was the bigger issue I had, but I figured a little whitespace in the modal below the gift certificate preview wouldn't hurt anything. 350px is taller than should ever be needed, but not too tall that it creates a breaking amount of space.

Reply:
https://github.com/davidjbradshaw/iframe-resizer Perhaps this might be of use for sizing the iframe without having to set hardcoded height and width values.